File: servicetaskmanager/src/main/java/com/example/alexeyglushkov/authtaskmanager/ServiceTask.java
package com.example.alexeyglushkov.authtaskmanager;

import com.example.alexeyglushkov.authorization.Auth.ServiceCommand;
import com.example.alexeyglushkov.taskmanager.task.Task;

import androidx.annotation.NonNull;

/**
 * Created by alexeyglushkov on 16.07.16.
 */
interface ServiceTask<T> extends ServiceCommand<T> {
    @NonNull Task getTask();
}

File: servicetaskmanager/src/main/java/com/example/alexeyglushkov/authtaskmanager/ServiceTask.kt
package com.example.alexeyglushkov.authtaskmanager

import com.example.alexeyglushkov.authorization.Auth.ServiceCommand
import com.example.alexeyglushkov.taskmanager.task.Task

/**
 * Created by alexeyglushkov on 16.07.16.
 */
internal interface ServiceTask<T> : ServiceCommand<T> {
    val task: Task
}